staging/src/k8s.io/kubectl/pkg/cmd/wait/wait.go

```diff
@@ -134,6 +135,7 @@ func (flags *WaitFlags) AddFlags(cmd *cobra.Command) {
 
 	cmd.Flags().DurationVar(&flags.Timeout, "timeout", flags.Timeout, "The length of time to wait before giving up.  Zero means check once and don't wait, negative means wait for a week.")
 	cmd.Flags().StringVar(&flags.ForCondition, "for", flags.ForCondition, "The condition to wait on: [delete|condition=condition-name].")
+	cmd.Flags().BoolVar(&flags.NonExistent, "nonexistent", flags.NonExistent, "The command should wait for a condition even if the object is nonexistent.")
```

Comment:
@rikatz  I don't think to add a new flag is the correct way to approach this problem. 

We can do something like this: 

If the resource doesn't exist, then the `wait` command should wait for some time ( default is 30s) and then time out.

Reply:
If you think that it is ok to change the default behaviour here - then take my 👍 here

Reply:
Will then think about a different approach :)

Maybe will rewrite a bunch of functions here, so I can pass the startTime to the inside functions IsDeletes and ConditionFn as well.

Reply:
@hpandeycodeit coming back into this. The default Timeout for the wait command is 7 days, so if I'm not adding a new flag (I agree we can reduce the qty of flags here) I'll wait 30s regardless of the timeout flag?

Reply:
Oh nevermind, I was looking into the wrong location :)